### PR TITLE
Pool management - Update block pool

### PR DESCRIPTION
**Pull request by GowthamShanmugam**
Signed-off-by: Gowtham Shanmugasundaram <gshanmug@redhat.com>

### CODE REVIEW

**Review by afreen23 — CHANGES_REQUESTED** [review]
change all form labels to sentence case as per console's ux guidelines.
i.e Pool Name  -> Pool name

**Comment by GowthamShanmugam** [comment]
@afreen23 changes done

**Comment by afreen23** [comment]
Please remove dependency commit 05a03ca801a46718be830f62897e287bfbfd13e3.
This commit needs approval console as well and can be handled in a separate PR.
With this Pr lets only have update logic

**Comment by GowthamShanmugam** [comment]
Dependent on: https://github.com/openshift/console/pull/8354

File: frontend/packages/ceph-storage-plugin/src/plugin.ts [hunk 1]
```diff
@@ -771,6 +771,23 @@ const plugin: Plugin<ConsumedExtensions> = [
...
+      callback: (kind, blockPoolConfig) => () => {
+        const blockPoolObject = { blockPoolConfig };
+        import('./components/modals/block-pool-modal/update-block-pool-modal')
+          .then((m) => m.updateBlockPoolModal(blockPoolObject))
+          .catch((e) => {
+            throw e;
+          });
+      },
+    },
+  },
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/plugin.ts:790]: Add this with the extension PR. Currently its causing test failure
```suggestion
```

**Review by afreen23 — CHANGES_REQUESTED** [review]

File: frontend/packages/ceph-storage-plugin/src/plugin.ts (hunk 1 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/plugin.ts:790]: if i add this due to import('./components/modals/block-pool-modal/update-block-pool-modal'), extension PR test will fail, as per hierarchy extension PR has to go first, so i can make test fail there

Comment (afreen23) [reply frontend/packages/ceph-storage-plugin/src/plugin.ts:790]: Okay, you can send another PR for enabling the edit action then.

**Comment by GowthamShanmugam** [comment]
@afreen @bipuladh Changes done, removed plugin changes

File: frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/modal-footer.tsx [hunk 2]
```diff
@@ -82,6 +91,8 @@ export const BlockPoolModalFooter = (props: BlockPoolModalFooterProps) => {
 type BlockPoolModalFooterProps = {
   state: BlockPoolState;
   dispatch: React.Dispatch<BlockPoolAction>;
-  onPoolCreation: (name: string) => void;
+  onPoolCreation?: (name: string) => void;
   onClick: () => void;
+  actionLabel: string;
+  closeLabel: string;
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/modal-footer.tsx:97]: nit: 
```suggestion
  onClick: () => void;
  actionLabel: string;
  closeLabel: string;
  onPoolCreation?: (name: string) => void;
```

File: frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/update-block-pool-modal.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  const MODAL_DESC = t(
+    'ceph-storage-plugin~A block pool is a logical entity providing elastic capacity to applications and workloads. Pools provide a means of supporting policies for access data resilience and storage efficiency.',
+  );
+  const MODAL_TITLE = t('ceph-storage-plugin~Edit Block Pool');
+
+  const [cephClusters, isLoaded, loadError] = useK8sWatchResource<CephClusterKind[]>(
+    cephClusterResource,
+  );
+  const cephCluster: CephClusterKind = useDeepCompareMemoize(cephClusters[0], true);
+
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/update-block-pool-modal.tsx:45]: nit:
```suggestion
  const [state, dispatch] = React.useReducer(blockPoolReducer, blockPoolInitialState);
  const [cephClusters, isLoaded, loadError] = useK8sWatchResource<CephClusterKind[]>(
    cephClusterResource,
  );
  
  const cephCluster: CephClusterKind = useDeepCompareMemoize(cephClusters[0], true);
const MODAL_DESC = t(
    'ceph-storage-plugin~A block pool is a logical entity providing elastic capacity to applications and workloads. Pools provide a means of supporting policies for access data resilience and storage efficiency.',
  );
  const MODAL_TITLE = t('ceph-storage-plugin~Edit Block Pool');

```

File: frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/update-block-pool-modal.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+      {
+        op: 'replace',
+        path: '/spec/parameters/compression_mode',
+        value: state.isCompressed ? COMPRESSION_ON : '',
+      },
+    ];
+
+    handlePromise(k8sPatch(CephBlockPoolModel, blockPoolConfig, patch), () => {
+      close();
+    });
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/update-block-pool-modal.tsx:102]: nit:
```suggestion
    handlePromise(k8sPatch(CephBlockPoolModel, blockPoolConfig, patch), () => close());
```

File: frontend/packages/ceph-storage-plugin/src/utils/block-pool.tsx [hunk 5]
```diff
@@ -66,6 +66,14 @@ export const PROGRESS_STATUS = (t: TFunction): ProgressStatusProps[] => [
     ),
     className: '',
   },
+  {
+    name: POOL_PROGRESS.NOTALLOWED,
+    icon: LockIcon,
+    desc: t(
+      'ceph-storage-plugin~Pool management tasks are not supported for default pool and openshift-container-storage external mode.',
```
Comment (afreen23) [inline frontend/packages/ceph-storage-plugin/src/utils/block-pool.tsx:73]: It wont be hypehened right ?
```suggestion
      'ceph-storage-plugin~Pool management tasks are not supported for default pool and Openshift Container Storage's  external mode.',
```

**Review by afreen23 — CHANGES_REQUESTED** [review]
Run yarn i18n and commit.

File: frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/modal-footer.tsx (hunk 2 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/modal-footer.tsx:97]: ack

File: frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/update-block-pool-modal.tsx (hunk 3 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/update-block-pool-modal.tsx:45]: ack

File: frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/update-block-pool-modal.tsx (hunk 4 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/components/modals/block-pool-modal/update-block-pool-modal.tsx:102]: ack

File: frontend/packages/ceph-storage-plugin/src/utils/block-pool.tsx (hunk 5 above)
Comment (GowthamShanmugam) [reply frontend/packages/ceph-storage-plugin/src/utils/block-pool.tsx:73]: ack

**Comment by GowthamShanmugam** [comment]
@afreen23 Changes done, all checks are passed

**Pull request merged**